var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -25,47 +25,152 @@ class Lammps(CMakePackage, CudaPackage):
 
     maintainers = ["rbberger"]
 
+    # rules for new versions and deprecation
+    # * new stable versions should be marked as preferred=True
+    # * a stable version that has updates and any of its outdated update releases should be
+    #   marked deprecated=True
+    # * only the latest update of any given stable release should be marked preferred=True
```

Comment:
Maybe move this to between lines 29 and 30?

https://spack.readthedocs.io/en/latest/packaging_guide.html#version-selection

Reply:
I realized that last "rule" created a problem with the sha256 for updates. Since I'm relying on the fact of `preferred=True` means to download the `stable_` tarballs, the checksum gets messed up if we then remove it later on (because that causes a `patch_` tarball to be downloaded).. So, for consistency, all stable versions (starting from 2020) should have `preferred=True`, independent of whether they are deprecated or not.